src/libsyntax/print/pp.rs

```diff
@@ -265,8 +265,8 @@ pub fn mk_printer<'a>(out: Box<io::Write+'a>, linewidth: usize) -> Printer<'a> {
 /// In this implementation (following the paper, again) the SCAN process is
 /// the method called 'pretty_print', and the 'PRINT' process is the method
 /// called 'print'.
-pub struct Printer<'a> {
-    pub out: Box<io::Write+'a>,
+pub struct Printer<W> {
+    pub out: W,
```

Comment:
In the past we've had some pretty bad code bloat problems by having a totally generic pretty printer, so I'm curious what instigated this change?